module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,60 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <iostream>
+#include <utility>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(
+    const std::array<std::array<uint8_t, width>, height>& bitmap) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressed{};
+    compressed.reserve(height * width);
+    int occurrences = 0;
+    uint8_t color_code = 0;
+    for (const auto& row : bitmap) {
+        color_code = *row.begin();
```

Comment:
why not simply front() ? It has not as clear as it can be why on the other hand you are using range loop (which is great) and later on you are taking the iterator.

Reply:
Thanks, done :)